front-end/generate-task-lists.js

```diff
@@ -62,13 +64,39 @@ const renderTaskList = event => {
 
             const markTaskAsDone = event => {
                 task.status = STATUSES.DONE;
-                renderDoneList();
-                renderTaskList();
+                return axios
+                    .post('/taskUpdated', {
+                        _id: task._id,
+                        status: task.status,
+                    })
+                    .then(response => {
+                        tasks.filter(task => {
```

Comment:
It's not a good use case for `.filter` since you are not filtering out any items from the array here (there's even no return statement). Use `.forEach` instead, but bear in mind that mutating an array is not a good idea in general (I don't want to introduce more complexity here)

Reply:
I can't imagine how to not mutate an array and at the same time change status of array's item. Can you give me some advices?

Reply:
I meant that you can mutate here, since an immutable solution is non-trivial (or requires an external package) and I would like to avoid more complex solutions in this case. Just use `.forEach` method instead of `.filter`. But remember that in general mutations are not optimal.